lib/js/patterns/filter-dropdown.js

```diff
@@ -0,0 +1,202 @@
+import bus from '../helpers/bus'
+
+import Guid from 'guid'
+
+// ┌─────────────────┐
+// │ Filter Dropdown │
+// └─────────────────┘
+// Select one or many from a searchable list
+
+function filterDropdown () {
+  bus.on('filterDropdown:bind', bindFilterDropdowns)
+  bus.on('filterDropdown:select', toggleItem)
+  bus.on('filterDropdown:select', emitActive)
+  bus.on('filterDropdown:select:remove', removeItem)
+  bus.on('filterDropdown:active', drawActive)
+  bus.on('filterDropdown:active:clear', clearActive)
+  bus.on('filterDropdown:toggle', toggleDropdown)
+  bus.on('filterDropdown:open', openList)
+  bus.on('filterDropdown:close', closeList)
+  bus.on('keyboard:escape', closeList)
+
+  function bindFilterDropdowns() {
+    var dropdowns = dom.findElements('.js-filter-dropdown');
+    dropdowns.forEach(function (dropdown) {
+      var dropdownId = dropdown.getAttribute('data-filter-dropdown')
+      var input = dropdown.querySelector('.filter-dropdown-input');
+      event.add(input, 'focus', inputFocus)
+
+      var opens = dropdown.querySelectorAll('.js-filter-dropdown-open')
+      for (let i = 0; i < opens.length; i++) {
+        var open = opens[i]
+        open.setAttribute('data-id', dropdownId)
+        event.add(open, event.click(), toggleClick);
+      }
+      var closes = dropdown.querySelectorAll('.js-filter-dropdown-close')
+      for (let i = 0; i < closes.length; i++) {
+        var close = closes[i]
+        close.setAttribute('data-id', dropdownId)
+        event.add(close, event.click(), toggleClick);
+      }
+
+
+      var items = dropdown.querySelectorAll('.filter-dropdown-link')
+      for (let i = 0; i < items.length; i++) {
+        var item = items[i];
+        item.setAttribute('data-item-id', i);
+        event.add(item, event.click(), itemClick);
+      }
+
+      event.add(input, 'keyup', function(e) {
+        var itemsArray = dom.nodeListToArray(items)
+        itemsArray.forEach(function(item) {
+          classy.add(item, 'hide')
+        })
+
+        dom.filterArray(input.value, itemsArray).forEach(function(item) {
+          classy.remove(item, 'hide')
+        })
+      })
+    })
+  }
+
+  function getOptions (e) {
+    var parent = dom.closest('js-filter-dropdown', e.target)
+    return {
+      parent: parent,
+      id: parent.getAttribute('data-filter-dropdown'),
+      item: e.target
+    }
+  }
+
+  function inputFocus (e) {
+    event.stopPropagation(e)
+    var options = getOptions(e)
+    bus.emit('filterDropdown:input:focus', options)
+  }
+
+  function itemClick (e) {
+    event.preventDefault(e)
+    event.stopPropagation(e)
+    var options = getOptions(e)
+    bus.emit('filterDropdown:select', options)
+  }
+
+  function toggleClick (e) {
+    var options = getOptions(e)
+    classy.toggle(e.target, 'is-active')
+    bus.emit('filterDropdown:toggle', options)
+  }
+
+  function toggleDropdown (options) {
+    var list = options.parent.querySelector('.filter-dropdown-list');
```

Comment:
not sure if we're using semicolons or not... it's pretty all over right now. We used to and it used to be linted?